mod-infra/src/main/scala/at/forsyte/apalache/infra/passes/WriteablePassOptions.scala

```diff
@@ -42,7 +42,12 @@ class WriteablePassOptions extends PassOptions {
     */
   def get[T](passName: String, optionName: String): Option[T] = {
     store.get(passName + "." + optionName) match {
-      case Some(value) => Some(value.asInstanceOf[T])
+      case Some(value: T) =>
+        Some(value)
+
+      case Some(value) =>
+        throw new IllegalArgumentException(s"Option $optionName has unexpected type: " + value.getClass)
```

Comment:
It is more of a convenience issue. In principle, the map stores objects of any kind. Since we do not want to do `isInstanceOf` all the time, this check is isolated in the method `Options.get[T]`. I guess, one can offload the whole work to the compiler. However, I am not sure it is worth the effort in this case.